pkg/controller/common/helpers.go

```diff
@@ -1397,3 +1399,21 @@ func GetCAsFromConfigMap(cm *corev1.ConfigMap, key string) ([]byte, error) {
 	}
 	return nil, fmt.Errorf("%s not found in %s/%s", key, cm.Namespace, cm.Name)
 }
+
+// Determines if an on-cluster layering rebuild is required for the changes
+func RequiresRebuild(oldMC, newMC *mcfgv1.MachineConfig) bool {
+	return oldMC.Spec.OSImageURL != newMC.Spec.OSImageURL ||
+		oldMC.Spec.KernelType != newMC.Spec.KernelType ||
+		!unorderedEqual(oldMC.Spec.Extensions, newMC.Spec.Extensions) ||
+		!unorderedEqual(oldMC.Spec.KernelArguments, newMC.Spec.KernelArguments)
+}
+
+// Determines if changes can be safely applied without node disruption
+func CanBeLiveApplied(oldMC, newMC *mcfgv1.MachineConfig) bool {
+	return !RequiresRebuild(oldMC, newMC)
+}
+
+// Helper for unordered slice comparison (set equality)
+func unorderedEqual(a, b []string) bool {
```

Comment:
issue: Once removed from the `RequiresRebuild()` function, it can be dropped from here too.